packages/config/src/files.js

```diff
@@ -76,31 +79,30 @@ const checkBuildDir = async function (buildDir, repositoryRoot) {
 
 // Some configuration properties have default values that are only set if a
 // specific directory/file exists in the build directory
-const addDefaultPaths = async function (build, baseRel) {
+const addDefaultPaths = async function (config, baseRel) {
   const props = await Promise.all(
-    DEFAULT_PATHS.map(({ property, defaultPath }) => addDefaultPath({ build, property, baseRel, defaultPath })),
+    DEFAULT_PATHS.map(({ defaultPath, object }) => addDefaultPath({ baseRel, defaultPath, object })),
   )
-  return Object.assign({}, build, ...props)
+
+  return deepMerge(...props, config)
 }
 
 const DEFAULT_PATHS = [
-  // @todo Remove once we drop support for the legact default functions directory.
-  { property: 'functions', defaultPath: 'netlify-automatic-functions' },
```

Comment:
Some users [are still using `netlify-automatic-functions` in production](https://cloud.us.humio.com/netlify-us-production/search?query=%22netlify-automatic-functions%22&live=false&start=7d&fullscreen=false), we might want to keep support there?

Reply:
Ah, good point. I misunderstood the scope of https://github.com/netlify/build/pull/2469, and I thought nobody was using this anymore. I'll reinstate it. Thanks!

Reply:
Done in https://github.com/netlify/build/pull/2496/commits/69a6d95a65dd511b8d8f732d684edd910a040c35. I'm using `functionsDirectoryOrigin: "default-v1"` for this case, just like in https://github.com/netlify/build/pull/2496#discussion_r603935754.

Reply:
:+1: 

Related: https://github.com/netlify/build/issues/2252#issuecomment-802156510